pgjdbc/src/main/java/org/postgresql/Driver.java

```diff
@@ -640,15 +640,20 @@ private static String database(Properties props) {
    * @return the timeout from the URL, in milliseconds
    */
   private static long timeout(Properties props) {
+    int result = 0;
     String timeout = PGProperty.LOGIN_TIMEOUT.get(props);
     if (timeout != null) {
       try {
-        return (long) (Float.parseFloat(timeout) * 1000);
+        result = Integer.parseInt(timeout);
```

Comment:
Hi vlsi, I explained the reason in original issue #879 comments. If you still think we shouldn't change, then I will revert it back.